src/tape/icp.jl

```diff
@@ -106,16 +106,47 @@ function reverse_pass!(tape::InstructionTape, input::AbstractArray)
         t = tape
         t[i].output.value = t[i].output.value ∩ t[i].cache
         op = IntervalContractors.reverse_operations[Symbol(t[i].func)]
+        if op == :mul_rev
+            if istracked(t[i].input[1]) && istracked(t[i].input[2])
+                rev_result = mul_rev_1(value(t[i].output), value.(t[i].input)...), mul_rev_2(value(t[i].output), value.(t[i].input)...)
+                t[i].input[1].value = rev_result[2]
+                if 0 < t[i].input[1].index <= n
+                    input[t[i].input[1].index] = rev_result[2]
+                end
+                t[i].input[2].value = rev_result[1]
+                if 0 < t[i].input[2].index <= n
+                    input[t[i].input[2].index] = rev_result[1]
+                end
+            elseif istracked(t[i].input[1])
+                rev_result = mul_rev_2(value(t[i].output), value.(t[i].input)...)
+                t[i].input[1].value = rev_result
+                if 0 < t[i].input[1].index <= n
```

Comment:
What is this line for?

Reply:
If the index of the variables is between 0 and n, that means that the variable is one of the input variables, for which, the changes need to be reflected in the input array as well.